src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -126,4 +133,85 @@ public static void CheckZIP64VersionIsSet_ForSmallFilesAfterBigFiles(bool isComp
             File.Delete(zipArchivePath);
         }
     }
+
+    [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized), nameof(PlatformDetection.Is64BitProcess))] // don't run it on slower runtimes
+    [OuterLoop("It requires around 6 GB of free disk space")]
+    public static void CompatZip64BeforeAndAfterFix()
+    {
+        // This test has the purpose of confirming that ZipArchive can still process a zip file that was created
+        // with these APIs before the Zip64 bug was fixed: https://github.com/dotnet/runtime/pull/102053
+
+        ushort buggyZip64Version = 20;
+        byte[] largeBuffer = GC.AllocateUninitializedArray<byte>(1_000_000_000); // 1 GB
+        string zipArchivePath = Path.Combine(Path.GetTempPath(), "over4GB.zip");
+
+        try
+        {
+            using FileStream fs = File.Open(zipArchivePath, FileMode.Create, FileAccess.ReadWrite);
+
+           // Create
+            using (ZipArchive archive = new(fs, ZipArchiveMode.Create, true))
+            {
+                ZipArchiveEntry file = archive.CreateEntry("file.txt", CompressionLevel.NoCompression);
+
+                using (Stream stream = file.Open())
+                {
+                    // Write 5GB of data
+                    for (var i = 0; i < 5; i++)
+                    {
+                        stream.Write(largeBuffer);
+                    }
+                }
+            }
+            Assert.True(fs.Length > int.MaxValue, $"File size is not big enough to test the Zip64 fix: {fs.Length} vs {int.MaxValue}");
+
+            fs.Position = 0;
+
+            // Open archive to modify the bit as it used to look before fix
```

Comment:
I noticed the source has a define for DEBUG_FORCE_ZIP64.  I don't see us ever use that, but I imagine the idea was we could use reflection to set that field and it would make a smaller zip use Zip64.  That might be an interesting method to test this without forcing things to outerloop.

Reply:
Good to keep it in mind. I wasn't aware that it could be used for testing. At the moment I felt that it was more important to confirm that the real life scenario is working. I'll consider using the define in the future.